addons/mail/static/src/core/common/suggestion_service.js

```diff
@@ -219,33 +231,45 @@ export class SuggestionService {
                 return persona.type === "partner";
             });
         }
-        return partners;
+        return [...partners];
+    }
+
+    /**
+     * @param {String} term
+     * @param {object[]} suggestions
+     * @param {string[]} fields
+     */
+    searchByTerm(term, suggestions, fields) {
+        return suggestions.filter((suggestion) =>
+            fields.some((field) => suggestion[field] && cleanTerm(suggestion[field]).includes(term))
+        );
+    }
+
+    filterPartnerSugestions(partners) {
+        return partners.filter((partner) => partner.name);
+    }
+
+    filterSpecialSuggestions(term, suggestions, thread) {
+        return suggestions.filter(
+            (special) =>
+                thread &&
+                special.channel_types.includes(thread.channel_type) &&
+                term.length >= Math.min(4, special.label.length)
+        );
     }
 
     searchPartnerSuggestions(cleanedSearchTerm, thread, sort) {
         const partners = this.getPartnerSuggestions(thread);
-        const suggestions = [];
-        for (const partner of partners) {
-            if (!partner.name) {
-                continue;
-            }
-            if (
-                cleanTerm(partner.name).includes(cleanedSearchTerm) ||
-                (partner.email && cleanTerm(partner.email).includes(cleanedSearchTerm))
-            ) {
-                suggestions.push(partner);
-            }
-        }
-        suggestions.push(
-            ...this.store.specialMentions.filter(
-                (special) =>
-                    thread &&
-                    special.channel_types.includes(thread.channel_type) &&
-                    cleanedSearchTerm.length >= Math.min(4, special.label.length) &&
-                    (special.label.startsWith(cleanedSearchTerm) ||
-                        cleanTerm(special.description.toString()).includes(cleanedSearchTerm))
-            )
+        const partnerSuggestions = this.filterPartnerSugestions(partners);
+        const specialSuggestions = this.filterSpecialSuggestions(
+            cleanedSearchTerm,
+            this.store.specialMentions,
+            thread
```

Comment:
I'm not fond of function named "filter" in the suggestion service with so many params.

What about `thread.getSpecialSuggestion(searchTerm)`? Intent of function is easier to grasp, single param, and global data are just retrieved from `store`.